docs/cli.md

```diff
@@ -580,6 +580,8 @@ poetry config [options] [setting-key] [setting-value1] ... [setting-valueN]
 `setting-key` is a configuration option name and `setting-value1` is a configuration value.
 See [Configuration]({{< relref "configuration" >}}) for all available settings.
 
+{{% warning %}} Use `--` to terminate option parsing in the traditional Unix tradition else, commands like `poetry config http-basic.custom-repo gitlab-ci-token ${GITLAB_JOB_TOKEN}` will fail if `${GITLAB_JOB_TOKEN}` starts with a hyphen (`-`). {{% /note %}}
+
```

Comment:
```suggestion
{{% warning %}}
Use `--` to terminate option parsing, otherwise commands like
`poetry config http-basic.custom-repo gitlab-ci-token ${GITLAB_JOB_TOKEN}`
will fail if `${GITLAB_JOB_TOKEN}` starts with a hyphen (`-`).
{{% /warning%}}

```